nuxeo-runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/services/config/ConfigurationServiceImpl.java

```diff
@@ -35,7 +45,41 @@
 
     public static final String CONFIGURATION_EP = "configuration";
 
-    protected ConfigurationPropertyRegistry registry = new ConfigurationPropertyRegistry();
+    public static final String COMPONENT_NAME = "org.nuxeo.runtime.config";
+
+    protected ObjectMapper mapper = null;
+
+    public ConfigurationServiceImpl() {
+        mapper = new ObjectMapper();
+    }
+
+    /**
+     * XXX remove once we are able to get such a cached map from DefaultComponent
+     *
+     * @since 10.3
+     */
+    protected Map<String, ConfigurationPropertyDescriptor> descriptors = null;
+
+    /**
+     * XXX remove once we are able to get such a cached map from DefaultComponent.
+     * <p>
+     * We'd ideally need a <T extends Descriptor> Map<String, T> getDescriptors(String xp) with cache method.
+     *
+     * @since 10.3
+     */
+    protected Map<String, ConfigurationPropertyDescriptor> getDescriptors() {
+        if (descriptors == null) {
+            synchronized (this) {
+                if (descriptors == null) {
+                    List<ConfigurationPropertyDescriptor> descs = getDescriptors(CONFIGURATION_EP);
+                    descriptors = descs.stream().collect(Collectors.toMap(desc -> desc.getId(), desc -> desc));
+                    return descriptors;
+                }
+            }
+
+        }
```

Comment:
This has to be handled at lower level, if needed a cache will be implemented in DescriptorRegistry, could you remove your cache here ?
Temporary things tend to become permanent

Reply:
yes, we can but I'd rather prefer you do the modification once you implement the cache. We have critical higher level features, bug fixes, improvements, etc. relying on this and I wish it shouldn't take that long to have them done ;)

Reply:
anyway, I need `Map<String, ConfigurationPropertyDescriptor> getDescriptors()` method